### PR TITLE
Happy Project - Moa Hammarbäck 

### PR DESCRIPTION
**Pull request by mhammarback**
Link to my deployed project: https://happy-booth-d6f66f.netlify.app/

My reflections on how this week's project turned out:
This week was fun but hard, i do however really enjoy React. 
I had the most issues with the like button but in the end i manage to make it work. 
I also feel that i have trouble with indentation when it comes to javascript so i need to work on that. 

Things I'd like to have clarified or explained in more detail:
Without Jennies video from the previous class i am not sure i would have made the like button work actually so i am really happy that we have more material to watch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+
+The project was made using React and the main goal was to practice fetcing data from the Happy thoughts API and using both useState and useEffect React Hooks.
 
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
 
 ## The problem
 
-Describe how you approached to problem, and what tools and techniques you used to solve it. How did you plan? What technologies did you use? If you had more time, what would be next?
+I started with a mobile first approach and built a React skelaton. My app contains a  main Component, a HappyForm component where you can post new messages and a HappyThoughtList component where i display the older messages and also where you can like other peoples messages.  
+
```
Comment (sofiavazs) [inline README.md:12]: Nice job making it responsive! looks perfect! 👏🏻

File: code/README.md [hunk 2]
```diff
@@ -1,19 +1,33 @@
-# Technigo React Starter App
+# Happy Thoughts
 
-This app has been generated using `create-react-app`, then cleaned up a little so we have a project with just the elements we need to get started.
+This project was made during the Technigo Bootcamp.
+The app is similar to Twitter but with only happy thougths!
```
Comment (sofiavazs) [inline code/README.md:4]: 💌

File: code/src/App.js [hunk 3]
```diff
@@ -1,9 +1,12 @@
-import React from 'react'
+import React from "react";
+
+import Main from './Main';
+
```
Comment (sofiavazs) [inline code/src/App.js:5]: Awesome job separating local dependencies from the outer dependencies 💃

File: code/src/HappyForm.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+
+
```
Comment (sofiavazs) [inline code/src/HappyForm.js:3]: maybe you can remove this line?

File: code/src/HappyForm.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+const HappyForm = ({ onMessageChange }) => {
+    const [message , setMessage] = useState ('');
+
```
Comment (sofiavazs) [inline code/src/HappyForm.js:6]: same here ;)

File: code/src/HappyForm.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    const MESSAGES_URL ="https://happy-thoughts-technigo.herokuapp.com/thoughts";
```
Comment (sofiavazs) [inline code/src/HappyForm.js:7]: Really nice that you've assigned the url to a variable!

File: code/src/HappyForm.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+     fetch(MESSAGES_URL, {
+        method: "POST",
+        headers:
+        {"Content-Type":"application/json"},
+        body:JSON.stringify({message: message})
+        }
+        ).then(() => {
+            window.location.reload();
+        })
+        .catch(err => console.log("error:", err))
```
Comment (sofiavazs) [inline code/src/HappyForm.js:23]: Nice!

File: code/src/HappyForm.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+        
+    };
+
+    return (
+    <>
+        <form onSubmit = {handleFormSubmit}>
+            <section className ="form-conatiner">
```
Comment (sofiavazs) [inline code/src/HappyForm.js:30]: I noticed you have a typo in "form-container" maybe that will bring you future unnecessary frustrations, so keep that in mind ;p

File: code/src/HappyForm.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+                Send Happy Thought
+
+                <span className="green-heart"
+                role="img"
+                aria-label="Heart">
+                {" 💚 "}
+                </span>
+                </button>
+
```
Comment (sofiavazs) [inline code/src/HappyForm.js:62]: I know you did mention you have a bit of trouble with indentation, but the attributes should be one step to the right. You also have extra lines, for ex line 53 and 55.

File: code/src/HappyForm.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+                <p className="thoughts-length-left"> {message.length} / 140 </p>
```
Comment (sofiavazs) [inline code/src/HappyForm.js:63]: nice that you made a character counter!

File: code/src/HappyThougthList.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import React  from "react";
+
```
Comment (sofiavazs) [inline code/src/HappyThougthList.js:2]: you could remove this line, perhaps?

File: code/src/HappyThougthList.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+                    <span className="heart-button-icon" role="img" aria-label="Heart">
+                    {" 💚"} 
+                    </span>
+                    </button>
+                    x {hearts}
+                </p>
+
+                <span className ="message-time">
+                    {moment(createdAt).fromNow()}; 
+                </span>
```
Comment (sofiavazs) [inline code/src/HappyThougthList.js:36]: indentation should be one step to the right over here too ;p

File: code/src/Main.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
+import React, { useState, useEffect } from "react";
+
+import HappyForm from "./HappyForm"
+import HappyThougthList from "./HappyThougthList";
+
+
+
```
Comment (sofiavazs) [inline code/src/Main.js:7]: too many lines over here maybe?

File: code/src/Main.js [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+const Main = () => { 
+const [thoughts, setThoughts] = useState([]); 
+
+const THOUGTHS_URL ="https://happy-thoughts-technigo.herokuapp.com/thoughts";
+
+    useEffect(() => {
+        fetchThougths();
+    }, []);  
+
```
Comment (sofiavazs) [inline code/src/Main.js:16]: I've noticed that you're not very consistent with semi colons, maybe you can decide either to use it everywhere or nowhere ;p

File: code/src/index.css [hunk 15]
```diff
@@ -11,3 +11,155 @@ code {
...
+  .previous-messages {
+    width: 60%;
+  }
+
+  .form-button {
+  width: 40%;
+  font-size: 16px;
+  padding: 15px 32px;
+  }
+ }
```
Comment (sofiavazs) [inline code/src/index.css:165]: Nicely done with responsiveness ✨ you can fix indentation on the css properties as well, one step to the right

**Review by sofiavazs — APPROVED** [review]
Great job Moa! Love the green! 

You achieved the blue and even did some stretch goals from the red level, nicely done! 
I did make some comments along the way, have a look at those and hope it will help for future projects! Overall you did a great job and should be really proud! 🥳

File: code/src/HappyForm.js [hunk 16]
```diff
@@ -0,0 +1,70 @@
+import React, { useState } from "react";
+
+const HappyForm = ({ onMessageChange }) => {
+    const [message , setMessage] = useState ('');
+    const MESSAGES_URL ="https://happy-thoughts-technigo.herokuapp.com/thoughts";
+
+    const handleFormSubmit = (event) => {
+    event.preventDefault(); 
+    onMessageChange(setMessage);
```
Comment (EmOlsz) [inline code/src/HappyForm.js:9]: Wrong indentation

**Pull request closed without merging**